Cython/Utility/Overflow.c

```diff
@@ -361,7 +361,7 @@ static CYTHON_INLINE {{TYPE}} __Pyx_{{BINOP}}_{{NAME}}_checking_overflow({{TYPE}
             return ({{TYPE}}) __Pyx_{{BINOP}}_unsigned_long_long_checking_overflow(a, b, overflow);
 #endif
         } else {
-            abort(); return 0; /* handled elsewhere */
+            Py_FatalError("Please report this as a bug in Cython - this code path should not be executed"); return 0; /* handled elsewhere */
```

Comment:
If this case really ever happens and we output this message, this will be the unhelpful bug report that we get. We'd help ourselves if we put at least some relevant information in there. The name of the type that we're trying to map here should be easy to add, the sizes might be difficult.